File: 1290-convert-binary-number-in-a-linked-list-to-integer/1290-convert-binary-number-in-a-linked-list-to-integer.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */

int getDecimalValue(struct ListNode* head){
    int i = 0;
    int k = 0;
    int exp;
    int a[30];  //2^30
    
    while(head != NULL)
    {
        a[i++] = head->val;
        head = head->next;
    }
    
    i--;    
    exp = 0;
    while(i >= 0)
    {
        if(a[i] == 1)
            k += a[i] * pow(2,exp);
        exp++;
        i--;
    }
    return k;
}

File: 1290-convert-binary-number-in-a-linked-list-to-integer/1290-convert-binary-number-in-a-linked-list-to-integer.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head) {
        int value = 0;
        int len = 0;
        ListNode tmp = head;
        while (tmp != null){
            tmp = tmp.next;
            len++;
        }
        while(len > 0){
            value += Math.pow(2, len-1) * head.val;
            head = head.next;
            len--;
        }
        return value;
    }
}